pallets/gear/src/benchmarking/tests/lazy_pages.rs

```diff
@@ -0,0 +1,497 @@
+                GranularityPage::from_offset(addr as u32),
+                GranularityPage::from_offset(addr as u32 + access_size - 1),
+            ]
+            .into_iter()
+            .collect();
+            if rng.gen_bool(LOAD_PROB) {
+                instrs.push(Instruction::I32Const(addr));
+                instrs.push(Instruction::I32Load(2, 0));
+                instrs.push(Instruction::Drop);
+
+                for page in accessed_pages {
+                    if !write_pages.contains(&page) && !write_after_read_pages.contains(&page) {
+                        read_pages.insert(page);
+                    }
+                }
+            } else {
+                instrs.push(Instruction::I32Const(addr));
+                instrs.push(Instruction::I32Const(u32::MAX as i32));
+                instrs.push(Instruction::I32Store(2, 0));
+
+                for page in accessed_pages {
+                    if !write_pages.contains(&page) {
+                        if read_pages.contains(&page) {
+                            write_after_read_pages.insert(page);
+                        } else if !write_after_read_pages.contains(&page) {
+                            write_pages.insert(page);
+                        }
+                    }
+                }
+            }
+        }
+
+        let code = WasmModule::<T>::from(ModuleDefinition {
+            memory: Some(ImportedMemory::max::<T>()),
+            handle_body: Some(body::from_instructions(instrs)),
+            ..Default::default()
+        });
+        let instance = Program::<T>::new(code, vec![]).unwrap();
+        let source = instance.caller.into_origin();
+        let program_id = ProgramId::from_origin(instance.addr);
+
+        // Append data in storage for some pages.
+        for page in (0..rng.gen_range(0..MAX_PAGES_WITH_DATA))
+            .map(|_| GranularityPage::new(rng.gen_range(0..size_psg.raw())).unwrap())
+        {
+            for page in page.to_pages_iter::<PageNumber>() {
+                ProgramStorageOf::<T>::set_program_page_data(
+                    program_id,
+                    page,
+                    PageBuf::new_zeroed(),
+                );
+            }
+        }
+
+        let exec = common_utils::prepare_exec::<T>(
+            source,
+            HandleKind::Handle(program_id),
+            vec![],
+            0..0,
+            Default::default(),
+        )
+        .unwrap();
+
+        let charged: Vec<(u64, u64)> = (0..2)
+            .map(|_i| {
+                let mut exec = exec.clone();
+                let weights = LazyPagesWeights {
+                    read: rng.gen_range(0..MAX_COST),
+                    write: rng.gen_range(0..MAX_COST),
+                    write_after_read: rng.gen_range(0..MAX_COST),
+                };
+                exec.block_config.pages_config.lazy_pages_weights = weights.clone();
+
+                let charged_for_pages = weights.read * gear_in_psg as u64 * read_pages.len() as u64
```

Comment:
mb saturating with unwraps? these tests run in release mode so overflow checks are excluded

Reply:
done